src/libraries/Common/src/System/Security/Cryptography/X509Certificates/X509CertificateLoader.Pkcs12.cs

```diff
@@ -375,7 +375,18 @@ private static void ProcessSafeContents(
 
                     if (ProviderNameIsRelevant)
                     {
-                        if (!loaderLimits.PreserveStorageProvider)
+                        const X509KeyStorageFlags EphemeralKeySet =
+#if NET
+                            X509KeyStorageFlags.EphemeralKeySet;
+#else
+                            (X509KeyStorageFlags)0x20;
+#endif
+
+                        if ((bagState.StorageFlags & EphemeralKeySet) != 0)
+                        {
+                            // Ephemeral loads have to go into CNG, but also won't get access denied.
```

Comment:
I can't say I am used to seeing many empty if statements. Why not write the second as `if ((bagState.StorageFlags & EphemeralKeySet) == 0 && !loaderLimits.PreserveStorageProvider)`?